src/libsystemd/sd-netlink/netlink-message.c

```diff
@@ -37,6 +37,8 @@
 #define GET_CONTAINER(m, i) ((i) < (m)->n_containers ? (struct rtattr*)((uint8_t*)(m)->hdr + (m)->containers[i].offset) : NULL)
 #define PUSH_CONTAINER(m, new) (m)->container_offsets[(m)->n_containers++] = (uint8_t*)(new) - (uint8_t*)(m)->hdr;
 
+#define MAXIMUM_SOCKET_BUFFER_SIZE ((size_t)(sysconf(_SC_PAGESIZE) < 8192L ? sysconf(_SC_PAGESIZE) : 8192L))
```

Comment:
This limit is the same as in `libmnl`.

Reply:
And indeed this corresponds with how the kernel itself treats netlink. From `include/linux/netlink.h`:

```c
/*
 *      skb should fit one page. This choice is good for headerless malloc.
 *      But we should limit to 8K so that userspace does not have to
 *      use enormous buffer sizes on recvmsg() calls just to avoid
 *      MSG_TRUNC when PAGE_SIZE is very large.
 */
#if PAGE_SIZE < 8192UL
#define NLMSG_GOODSIZE  SKB_WITH_OVERHEAD(PAGE_SIZE)
#else
#define NLMSG_GOODSIZE  SKB_WITH_OVERHEAD(8192UL)
#endif
```

Reply:
can you add a comment here, what that is?

Hmm, this will evaluate sysconf() twice, I wonder if this shouldn't better be a caching function of some kind